drgn_tools/multipath.py

```diff
@@ -0,0 +1,345 @@
+            )
+            return
+        show_pgpaths(priority_group)
+
+
+def show_multipath(prog: Program, target: Object) -> None:
+    """
+    Show multipath properties.
+
+    :param prog: drgn Program
+    :param target: `struct dm_target *`
+
+    :returns: None:
+    """
+    multipath = cast("struct multipath *", target.private)
+    if multipath.current_pgpath:
+        current_dmpath = multipath.current_pgpath.path
+        curr_dev_name = current_dmpath.dev.name.string_().decode()
+    else:
+        curr_dev_name = "None"
+    if multipath.hw_handler_name:
+        hw_handler = multipath.hw_handler_name.string_().decode()
+    else:
+        hw_handler = "0"
+    if multipath.next_pg:
+        pg_num = multipath.next_pg.pg_num
+    elif multipath.current_pg:
+        pg_num = multipath.current_pg.pg_num
+    else:
+        pg_num = 1 if multipath.nr_priority_groups else 0
+    if has_member(multipath, "queue_mode"):
+        queue_mode = DM_QUEUE_MODE.get(multipath.queue_mode.value_(), "N/A")
+    else:
+        queue_mode = "-"
+    print("struct multipath * (%s)" % hex(multipath.value_()))
+    print("Current path = %s" % curr_dev_name)
+    print("hw_handler_name: %s" % hw_handler)
+    print("pg_num = %u" % pg_num)
+    print("queue_mode = %s" % queue_mode)
+    # print("Size(bytes) = %lu" % (target.len.value_() << SECTOR_SHIFT))
+    show_priority_groups(multipath)
+
+
+def iterate_targets(prog: Program, table: Object) -> None:
+    """
+    Iterate through targets.
+
+    :param prog: drgn Program
+    :param table: `struct dm_table *`
+
+    :returns: None
+    """
+    for index in range(0, table.num_targets):
+        target = table.targets[index]
+        show_multipath(prog, target)
+
+
+def dump_dm(
+    prog: Program, dm: Object, device_name: str, device_uuid: str
+) -> None:
+    """
+    :param prog: drgn Program
+    :param dm: `struct mapped_device *`
+    :param device_uuid: `char *`
+
+    :returns: None
+    """
+    if dm_target_name(dm) == "multipath":
+        print(
+            "%s Device [%d] %s" % (hyphen * 15, device_count[0], hyphen * 15)
+        )
+        disk_name = dm.disk.disk_name.string_().decode()
+        print(
+            "%s (%s) %s"
+            % (device_name, print_delim(device_uuid, "-"), disk_name)
+        )
+        iterate_targets(prog, dm_table(dm))
+        device_count[0] += 1
+
+
+def show_mp(prog: Program) -> None:
+    """
+    Dump multipath devices Info.
+    Iteratre through uuid rb tree/list
+    Iterate through name rb tree.
+
+    :param prog: drgn Program
+
+    :returns None:
+    """
+    ret = ensure_debuginfo(prog, ["dm_mod", "dm_multipath"])
+    if ret:
+        print(ret)
+        return
+
+    for dm, name, uuid in for_each_dm(prog):
+        dump_dm(prog, dm, name, uuid)
+
+
+class Multipath(CorelensModule):
+    """Display info about Multipath devices"""
+
+    name = "multipath"
+    debuginfo_kmods = ["dm_mod", "dm_multipath"]
```

Comment:
The other path selector are kernel modules like dm-round-robin/dm-service-time, i am wondering how corelens could work without knowing these module are required?

Reply:
I have added the path selection policies modules. 
`    debuginfo_kmods = ["dm_mod", "dm_multipath",
            "dm_service", "dm_queue_length",
            "dm_io_affinity", "dm_historical_service_time"]
`